From a485b6e60fc8365d3153f852cce714eb92201998 Mon Sep 17 00:00:00 2001
From: nxa15299 <Heinz.Wrobel@nxp.com>
Date: Mon, 5 Feb 2018 08:33:02 +0100
Subject: [PATCH] rcw.py: regexp parser was broken for write.b1.

As it turns out, it was not the command parsing for write that
was the problem. The generic parser was broken.
This should be fixed now.

Signed-off-by: nxa15299 <Heinz.Wrobel@nxp.com>
---
 rcw.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/rcw.py b/rcw.py
index cfdd323..cfe3371 100755
--- a/rcw.py
+++ b/rcw.py
@@ -284,7 +284,7 @@ def build_pbi(lines):
     for l in lines:
         # Check for an instruction without 0-2 parameters
         # The + ' ' is a hack to make the regex work for just 'flush'
-        m = re.search('([a-z]+)(|\.b1|\.b2|\.b4|\.short|\.long)\s*(?<=\s)([^,]*),?([^,]*),?([^,]*),?([^,]*)', l + ' ')
+        m = re.match(r'\s*([a-z]+)(|\.b1|\.b2|\.b4|\.short|\.long)\s*(?<=\s)([^,]*),?([^,]*),?([^,]*),?([^,]*)', l + ' ')
         if not m:
             print 'Unknown PBI subsection command "%s"' % l
             return ''
@@ -415,7 +415,7 @@ def parse_source_file(source):
             continue
 
         # Is it a subsection?
-        m = re.search(r'\.([a-zA-Z]+)', l)
+        m = re.match(r'\.([a-zA-Z]+)', l)
         if m:
             if in_subsection:
                 in_subsection = False
@@ -432,14 +432,14 @@ def parse_source_file(source):
             continue
 
         # Is it an identifier?  %var=value
-        m = re.search(r'%([a-zA-Z]+[a-zA-Z0-9]+)=(.+)', l)
+        m = re.match(r'%([a-zA-Z]+[a-zA-Z0-9]+)=(.+)', l)
         if m:
             identifier, value = m.groups()
             vars[identifier] = value
             continue
 
         # Is it a single field definition?  NAME[position]
-        m = re.search(r'([A-Z0-9_]+)\[([0-9a-zA-Z]+)]', l)
+        m = re.match(r'([A-Z0-9_]+)\[([0-9a-zA-Z]+)]', l)
         if m:
             name, position = m.groups()
             position = int(position, 0)
@@ -449,7 +449,7 @@ def parse_source_file(source):
             continue
 
         # Is it a ranged field definition?  NAME[begin:end]
-        m = re.search(r'([A-Z0-9_]+)\[([0-9a-zA-Z]+):([0-9a-zA-Z]+)\]', l)
+        m = re.match(r'([A-Z0-9_]+)\[([0-9a-zA-Z]+):([0-9a-zA-Z]+)\]', l)
         if m:
             (name, begin, end) = m.groups()
             begin = int(begin, 0)
@@ -459,7 +459,7 @@ def parse_source_file(source):
             continue
 
         # Is it a field assignment? NAME=value
-        m = re.search(r'([A-Z0-9_]+)=([0-9a-zA-Z]+)', l)
+        m = re.match(r'([A-Z0-9_]+)=([0-9a-zA-Z]+)', l)
         if m:
             (name, value) = m.groups()
             value = int(value, 0)
-- 
2.15.1

